From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Add 'gtk' USE flag, loosen all constraints. Does not build with gi-gtk-4.

diff -urN Gifcurry-3.0.0.1/Gifcurry.cabal Gifcurry-3.0.0.1-r1/Gifcurry.cabal
--- Gifcurry-3.0.0.1/Gifcurry.cabal	2022-02-21 14:11:07.446562182 -0700
+++ Gifcurry-3.0.0.1-r1/Gifcurry.cabal	2022-02-22 13:56:59.606603665 -0700
@@ -35,36 +35,42 @@
 library
   exposed-modules:    Gifcurry
   build-depends:        base >= 4.7 && < 5
-                      , process >= 1.2 && <= 1.4.4
-                      , temporary >= 1.2 && < 1.3
-                      , directory == 1.3.*
-                      , text == 1.2.*
-                      , filepath == 1.4.*
+                      , process >= 1.2
+                      , temporary >= 1.2
+                      , directory >= 1.3
+                      , text >= 1.2
+                      , filepath >= 1.4
   hs-source-dirs:       ./src
                       , ./src/lib/
   ghc-options:        -Wall -freverse-errors
   default-language:   Haskell2010
 
+flag gtk
+  description: Build the gifcurry_gui executable
+  default: False
+
 executable            gifcurry_gui
+  if !flag(gtk)
+    buildable: False
   main-is:            Main.hs
-  build-depends:        base == 4.9.*
-                      , haskell-gi == 0.21.*
-                      , haskell-gi-base == 0.21.*
-                      , gi-gobject == 2.0.*
-                      , gi-glib == 2.0.*
-                      , gi-gdk == 3.0.*
-                      , gi-gtk == 3.0.*
-                      , gi-cairo == 1.0.*
-                      , gi-gst == 1.0.*
-                      , gi-gstvideo == 1.0.*
-                      , cairo == 0.13.*
-                      , bytestring == 0.10.*
-                      , process >= 1.2 && <= 1.4.4
-                      , temporary >= 1.2 && < 1.3
-                      , directory == 1.3.*
-                      , text == 1.2.*
-                      , filepath == 1.4.*
-                      , transformers == 0.5.*
+  build-depends:        base
+                      , haskell-gi >= 0.21
+                      , haskell-gi-base >= 0.21
+                      , gi-gobject >= 2.0
+                      , gi-glib >= 2.0
+                      , gi-gdk >= 3.0 && < 4
+                      , gi-gtk >= 3.0 && < 4
+                      , gi-cairo >= 1.0
+                      , gi-gst >= 1.0
+                      , gi-gstvideo >= 1.0
+                      , cairo >= 0.13
+                      , bytestring >= 0.10
+                      , process >= 1.2
+                      , temporary >= 1.2
+                      , directory >= 1.3
+                      , text >= 1.2
+                      , filepath >= 1.4
+                      , transformers >= 0.5
   other-modules:        Paths_Gifcurry
                       , GuiRecords
                       , Gifcurry
@@ -83,12 +89,12 @@
 executable            gifcurry_cli
   main-is:            Main.hs
   build-depends:        base >= 4.7 && < 5
-                      , process >= 1.2 && <= 1.4.4
-                      , temporary >= 1.2 && < 1.3
-                      , directory == 1.3.*
-                      , cmdargs == 0.10.*
-                      , text == 1.2.*
-                      , filepath == 1.4.*
+                      , process >= 1.2
+                      , temporary >= 1.2
+                      , directory >= 1.3
+                      , cmdargs >= 0.10
+                      , text >= 1.2
+                      , filepath >= 1.4
   other-modules:      Gifcurry
   ghc-options:        -Wall -freverse-errors
   hs-source-dirs:       ./src/
